x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { CredentialsLogic } from './credentials_logic';
+import { ADMIN, PRIVATE } from '../../constants/credentials';
+
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: { http: { get: jest.fn(), delete: jest.fn() } } },
+}));
+import { HttpLogic } from '../../../shared/http';
+import { IApiToken, ICredentialsDetails } from '../../../../../common/types/app_search';
+import { IMeta } from '../../../../../common/types';
+
+function mount(defaults?: object) {
+  if (!defaults) {
+    resetContext({});
+  } else {
+    resetContext({
+      // the enterprise_search.app_search.credentials namespace is set in `path` in credentials_logic
+      defaults: {
+        enterprise_search: {
+          app_search: {
+            credentials: {
+              ...defaults,
+            },
+          },
+        },
+      },
+    });
+  }
+  CredentialsLogic.mount();
+}
+
+const newToken: IApiToken = {
+  id: 1,
+  name: 'myToken',
+  type: PRIVATE,
+  read: true,
+  write: true,
+  access_all_engines: true,
+};
+
+const credentialsDetails: ICredentialsDetails = {
+  lmAccount: {
+    id: '1',
+    key: '1234',
+  },
+  apiTokens: [],
+  apiUrl: 'http://example.com',
+  engines: [
+    { name: 'engine1', type: 'indexed', language: 'english', result_fields: [] },
+    { name: 'engine1', type: 'indexed', language: 'english', result_fields: [] },
+  ],
+};
+
+describe('CredentialsLogic', () => {
+  const DEFAULT_VALUES = {
+    activeApiToken: {
+      name: '',
+      type: PRIVATE,
+      read: true,
+      write: true,
+      access_all_engines: true,
+    },
+    activeApiTokenIsExisting: false,
+    activeApiTokenRawName: '',
+    apiTokens: [],
+    apiUrl: '',
+    dataLoading: true,
+    engines: [],
+    formErrors: [],
+    isCredentialsDataComplete: false,
+    isCredentialsDetailsComplete: false,
+    meta: null,
+    nameInputBlurred: false,
+    showCredentialsForm: false,
+  };
```

Comment:
Just thinking out loud - I wonder if we should DRY this out by switching to using Kea's `default` key instead of passing in defaults per-value. See the second code block in this doc example: https://kea.js.org/docs/guide/additional#defaults

Reply:
Interesting thought. Not sure which approach would be better.

Reply:
I tried it out the other day on a random new file and it wasn't totally working as expected / was giving me some strange errors. I might've been doing it wrong though 🤔 I'll chalk this up as a failed experiment for another day haha.